### ISSUE TITLE
ACS compilation doesn't update BEHAVIOR lump

### ISSUE THREAD

**Issue by esoeylemez** [sirjuddington/SLADE#847]
**SLADE Version**: 3.1.1.5 (00bae1f5ca76d6dd8775b35a9b2e0188b7b0146d), 3.1.2_b5 (d35095084c37c317b3edd47dfb2b343e80ccc5a0), master (f7409c504b40c4962f419038db934c32688ddd2e)
**Operating System**: Linux
**CMake flags**: -DNO_WEBVIEW=1

**Issue Details**:
When I tell SLADE to compile my SCRIPTS lump, according to the compiler log everything is fine, and it registers the BEHAVIOR lump as changed (it displays green or blue), but it remains empty (size is 0, and hex view shows nothing). This happens in WAD files, ZIP files and directories. I have tried with [ACC](https://github.com/rheit/acc/) and [BCC](https://github.com/wormt/bcc).

It doesn't seem to be related to my compiler setup, because if I make an intentional syntax error in the script, SLADE reports a compilation error as expected.

**Comment by esoeylemez** [comment]
I attempted to use the following workflow as a workaround, but it actually exhibits the same behaviour:

  * Use a directory project (File → Open directory),
  *

**Comment by esoeylemez** [comment]
I attempted to use the following workflow as a workaround, but it actually exhibits the same behaviour:

  * Use a directory project (File → Open directory),
  * Open a map within that directory,
  * Place the compiler output as `BEHAVIOR.o` in the project directory from outside,
  * SLADE registers that a file was added/changed externally and asks which changes to apply; apply all of them,
  * from the viewpoint of SLADE the file `BEHAVIOR.o` is empty (size 0, hex view shows nothing), even though it's definitely not empty,
  * copy/paste that file into the map. SLADE inserts an empty `BEHAVIOR` lump into the map, completely ignoring the file contents from the filesystem.

So I'm assuming that the bug isn't directly related to compilation, but something seems to go wrong with SLADE updating its internal project state.

Note that *importing* the compiler output into the map *does* work.

**Comment by esoeylemez** [comment]
Further investigation into this issue revealed that this seems to be related to my setting of the `TMPDIR` environment variable. This happens if I leave it unset, set it to the usual `/tmp` and a custom directory `~/tmp/slade`. However, if I set it to `/run/user/1000`, it seems to work. That directory is a `tmpfs`:

```
% mount | grep /run/user/1000
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=807012k,mode=700,uid=1000,gid=100)
```

Instead of errors SLADE seems to think that the file has size 0, which I have verified by monitoring the `size` argument in `ArchiveEntry::importFile`. In `/src/Archive/ArchiveEntry.cpp`, on line 491 (at f7409c504b40c4962f419038db934c32688ddd2e) it has a value of 0.

I hope this helps. In the meantime I'm going to set `TMPDIR` explicitly for SLADE.
